### PR TITLE
Fix race condition in test_pool_memmap_with_big_offset

### LINKED ISSUE #1040: Random timeout in test_pool_memmap_with_big_offset
Appears frequently on `windows` + `Python 3.8`.

### PR DESCRIPTION
**Pull request by pierreglaser**
Closes #1040

### CODE REVIEW

**Comment by pierreglaser** [comment]
test_pool_memmap_with_big_offset seems stable now, but the `pypy` failure keeps showing up and I cannot reproduce locally...

**Comment by pierreglaser** [comment]
@ogrisel WDYT?

File: joblib/test/test_memmapping.py [hunk 1]
```diff
@@ -521,9 +522,11 @@ def identity(arg):
...
 @with_numpy
 @with_multiprocessing
-@parametrize("factory", [MemmappingPool, _TestingMemmappingExecutor],
-             ids=["multiprocessing", "loky"])
-def test_pool_memmap_with_big_offset(factory, tmpdir):
+@parametrize(
+    "factory,retry_no",
+    list(itertools.product([MemmappingPool, _TestingMemmappingExecutor], range(3))),  # noqa
+    ids=['{}, {}'.format(x, y) for x, y in itertools.product(["multiprocessing", "loky"], map(str, range(3)))])  # noqa
+def test_pool_memmap_with_big_offset(factory, retry_no, tmpdir):
```
Comment (ogrisel) [inline joblib/test/test_memmapping.py:529]:
```suggestion
@parametrize(
    "factory,retry_no",
    list(itertools.product([MemmappingPool, _TestingMemmappingExecutor],
                           range(3))),
    ids=['{}, {}'.format(x, y) for x, y in itertools.product(["multiprocessing", "loky"],
                                                             map(str, range(3)))])
def test_pool_memmap_with_big_offset(factory, retry_no, tmpdir):
```

File: joblib/test/test_memmapping.py [hunk 2]
```diff
@@ -521,9 +522,13 @@ def identity(arg):
...
 @with_multiprocessing
-@parametrize("factory", [MemmappingPool, _TestingMemmappingExecutor],
-             ids=["multiprocessing", "loky"])
-def test_pool_memmap_with_big_offset(factory, tmpdir):
+@parametrize(
+    "factory,retry_no",
+    list(itertools.product([MemmappingPool, _TestingMemmappingExecutor],
+                           range(3))),
+    ids=['{}, {}'.format(x, y) for x, y in itertools.product(["multiprocessing", "loky"],
+                                                             map(str, range(3)))])
```
Comment (ogrisel) [inline joblib/test/test_memmapping.py:530]:
```suggestion
    ids=['{}, {}'.format(x, y) for x, y in itertools.product(
        ["multiprocessing", "loky"], map(str, range(3)))])
```

File: joblib/test/test_memmapping.py [hunk 3]
```diff
@@ -521,9 +522,13 @@ def identity(arg):
...
 @with_multiprocessing
-@parametrize("factory", [MemmappingPool, _TestingMemmappingExecutor],
-             ids=["multiprocessing", "loky"])
-def test_pool_memmap_with_big_offset(factory, tmpdir):
+@parametrize(
+    "factory,retry_no",
+    list(itertools.product([MemmappingPool, _TestingMemmappingExecutor],
+                           range(3))),
+    ids=['{}, {}'.format(x, y) for x, y in itertools.product(
+            ["multiprocessing", "loky"], map(str, range(3)))])
```
Comment (ogrisel) [inline joblib/test/test_memmapping.py:530]:
```suggestion
        ["multiprocessing", "loky"], map(str, range(3)))])
```

**Pull request merged**